plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -17,14 +19,14 @@ export interface GeneralArgs {
 
 export interface JobsQueryArgs {
   filter?: string | null;
-  sortBy?: SortOption | null;
-  sortDirection?: SortDirection | null;
+  sortBy?: SortOption;
```

Comment:
I agree with your point, but from my idea of how filters and sort work, you could specify no filter and no sorting. I would even like that because sort is expensive. 

In any case, I would not break compatibility, and would them do the types as you propose.

Reply:
We can guard filtering by checking against empty string

Reply:
I disagree, it is perfectly fine to have it undefined or null here, why would we want to force the user of the datalayer to always filter or sort?